### PR TITLE
Pathfinding

### PR DESCRIPTION
**Pull request by onaranjoUCM**

### CODE REVIEW

File: Drakhtar/GameObjects/Board.h [hunk 1]
```diff
@@ -38,6 +42,9 @@ class Board : public GameObject {
   // Checks if two cells are within a certain range of each other
   bool isInRange(Box *from, Box *to, int range);
 
+  // Checks if two cells are withing a certain range of each other considering pathfinding
```
Comment (MiguelZh) [inline Drakhtar/GameObjects/Board.h]: "withing" is spelled wrong, It should be "within".

File: Drakhtar/EventListeners/BoardController.cpp [hunk 2]
```diff
@@ -44,8 +44,9 @@ void BoardController::onClickStop(SDL_Point point) {
 
 void BoardController::onClickMove(Box* boxClicked) {
   // Checks if the box clicked is within movement range
-  if (board_->isInRange(activeUnit_->getBox(), boxClicked,
+  if (board_->isInMoveRange(activeUnit_->getBox(), boxClicked,
                         activeUnit_->getMoveRange())) {
+	board_->findPath(activeUnit_->getBox()->getIndex(), boxClicked->getIndex());
```
Comment (kyranet) [inline Drakhtar/EventListeners/BoardController.cpp]:
```suggestion
    board_->findPath(activeUnit_->getBox()->getIndex(), boxClicked->getIndex());
```

File: Drakhtar/third_party/AStar.cpp [hunk 3]
```diff
@@ -0,0 +1,178 @@
+// Copyright 2018 the daancode/a-star authors. All rights reserved. MIT license. (https://github.com/daancode/a-star)
```
Comment (kyranet) [inline Drakhtar/third_party/AStar.cpp]: Can you format everything, please? The linter will complain quite a lot.

Comment (kyranet) [reply Drakhtar/third_party/AStar.cpp]: Same for `AStar.h`

**Pull request merged**